onnxruntime/core/framework/execution_providers.h

```diff
@@ -76,6 +76,8 @@ class ExecutionProviders {
 
   bool Empty() const { return exec_providers_.empty(); }
 
+  size_t NumProviders() const { return exec_providers_.size(); }
```

Comment:
>NumProviders [](start = 9, length = 12)

maybe size() or count() matches better with begin()/end() below.

Reply:
The reason for 'begin' and 'end' is so you can use this class in a range based loop. Those functions don't match our naming convention though, but it's what C++ requires. Adding more things that don't match the naming convention starts to make it more confusing as to what the convention is, so I deliberately didn't use 'size' or 'count'. 

---
In reply to: [252394402](https://github.com/Microsoft/onnxruntime/pull/410#discussion_r252394402) [](ancestors = 252394402)